### PR TITLE
mac: Allow MAC scans while the rest of the Thread stack is disabled (#42)

### PR DESCRIPTION
**Pull request by jwhui**
This is a pull request for issue #42.

This change removes the enable/disable methods from the MAC interface and allows MAC actions to occur at any time, automatically disabling the radio when no other actions are required.

### CODE REVIEW

File: src/core/openthread.cpp [hunk 1]
```diff
@@ -112,6 +112,7 @@ const uint8_t *otGetExtendedPanId(void)
 void otSetExtendedPanId(const uint8_t *aExtendedPanId)
 {
     sThreadNetif->GetMac().SetExtendedPanId(aExtendedPanId);
+    sThreadNetif->GetMle().SetMeshLocalPrefix(aExtendedPanId);
```
Comment (darconeous) [inline src/core/openthread.cpp]: Is this change related to the scans? If not, it might make sense to split this change out into a separate commit.

Comment (jwhui) [inline src/core/openthread.cpp]: Split this change out into a separate commit, see #63.

**Comment by darconeous** [comment]
I still see commit cfd91ea in this branch. If you have #63, shouldn't that commit not be in this pull request?

**Comment by darconeous** [comment]
Otherwise, 👍

**Comment by jwhui** [comment]
Yes, I had branched off #63, rather than off master.  There is only one commit now after the merge and rebase.
